kind/pkg/build/node_image.go

```diff
@@ -41,7 +41,7 @@ type NodeImageBuildContext struct {
 
 // NewNodeImageBuildContext creates a new NodeImageBuildContext with
 // default configuration
-func NewNodeImageBuildContext(mode string) (ctx *NodeImageBuildContext, err error) {
+func NewNodeImageBuildContext(mode, imageName, baseImageName string) (ctx *NodeImageBuildContext, err error) {
```

Comment:
We may want to consider a struct to hold these config options... I'll leave it up to you whether we do this as part of this PR @BenTheElder

Reply:
I think 3 is probably still low enough that inline args is fine, if we get any more we can look at refactoring as we won't have downstream users of the library just yet, we'd break them when we move this out of t-i anyhow